core/app/models/spree/stock_item.rb

```diff
@@ -96,11 +96,23 @@ def process_backorders(number)
       end
 
       def conditional_variant_touch
+        variant.touch if inventory_cache_threshold.nil? || should_touch_variant?
+      end
+
+      def should_touch_variant?
         # the variant_id changes from nil when a new stock location is added
-        stock_changed = (count_on_hand_changed? && count_on_hand_change.any? { |c| c < 1 }) || variant_id_changed?
+        inventory_cache_threshold &&
+        (count_on_hand_changed? && count_on_hand_change.any? { |c| c < inventory_cache_threshold }) ||
+        variant_id_changed?
+      end
 
-        if !Spree::Config.binary_inventory_cache || stock_changed
-          variant.touch
+      def inventory_cache_threshold
+        # only warn if store is setting binary_inventory_cache (default = false)
+        @cache_threshold ||= if Spree::Config.binary_inventory_cache
```

Comment:
this was to avoid the deprecation warning being called every time we try to access that return value.

thoughts on this?